src/data/blob.rs

```diff
@@ -0,0 +1,420 @@
+// Copyright 2019 MaidSafe.net limited.
+//
+// This SAFE Network Software is licensed to you under the MIT license <LICENSE-MIT
+// https://opensource.org/licenses/MIT> or the Modified BSD license <LICENSE-BSD
+// https://opensource.org/licenses/BSD-3-Clause>, at your option. This file may not be copied,
+// modified, or distributed except according to those terms. Please review the Licences for the
+// specific language governing permissions and limitations relating to use of the SAFE Network
+// Software.
+
+use crate::{utils, Error, PublicKey, XorName};
+use bincode::serialized_size;
+use multibase::Decodable;
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+use std::{
+    fmt::{self, Debug, Formatter},
+    u64,
+};
+use tiny_keccak;
+
+/// Maximum allowed size for a serialised blob (ID) to grow to
+pub const MAX_BLOB_SIZE_IN_BYTES: u64 = 1024 * 1024 + 10 * 1024;
+
+/// PrivateBlob: an immutable chunk of data which can be deleted. Can only be fetched
+/// by the listed owner.
+#[derive(Hash, Eq, PartialEq, PartialOrd, Ord, Clone)]
+pub struct PrivateBlob {
+    /// Address.
```

Comment:
```suggestion
    /// Address. Omitted when serialising and calculated from the `value` and `owner` when
    /// deserialising.
```
I believe that it's nice to have this additonal info. Thoughts?

Reply:
Absolutely, I see that's what it says on PublicBlob. I think all comments in this file have been copied straight over. The name changes have been done with find and replace.